pytorch_lightning/core/lightning.py

```diff
@@ -1578,6 +1578,44 @@ def configure_gradient_clipping(self, optimizer, optimizer_idx, gradient_clip_va
             optimizer, gradient_clip_val=gradient_clip_val, gradient_clip_algorithm=gradient_clip_algorithm
         )
 
+    def lr_scheduler_step(
+        self,
+        scheduler: Any,
+        step: int,
```

Comment:
Personally, if I want to implement a custom scheduling mechanism I might want to have access to both the global_step and epoch.

Reply:
You have access through `self.trainer`.

Reply:
yes thought so.. but these are some of the arguments that actually takes part in scheduler step.
we have some others like:
```py
optimizer_step(epoch, ...) <- epoch can be accessed using `trainer.current_epoch`.
transfer_batch_to_device(..., device) <- device can be accessed using `self.device`.
```
since I personally know these are available, so not sure whether to remove them from user's point of view. but if we choose to remove them, we should update the above 2 hook as well IMO.

Reply:
Passing arguments that aren't used explicitly opens the door to the question of, which should be passed and which should not.

I also don't like having it in `optimizer_step`. But in the case of `transfer_batch_to_device`, it's good to have because the default implementation uses it.

Reply:
btw `transfer_batch_to_device` doesn't need one either. it's a lightningmodule, hook, device is a part of module itself.
for now, I'll remove it from `scheduler_step`, we can think of others later.